rustls/src/crypto/cipher.rs

```diff
@@ -270,52 +270,56 @@ const TLS12_AAD_SIZE: usize = 8 + 1 + 2 + 2;
 
 /// A key for an AEAD algorithm.
 ///
-/// This is a value type for a byte string up to `AeadKey::MAX_LEN` bytes in length.
-pub struct AeadKey {
-    buf: [u8; Self::MAX_LEN],
-    used: usize,
+/// This is a value type for a byte string of 128- or 256-bits.
+#[non_exhaustive]
+pub struct AeadKey(AeadKeyInner);
+
+#[non_exhaustive]
```

Comment:
Don't think these need `#[non_exhaustive]` -- the former because its fields are private anyway, the latter because the type itself is private.